### PR TITLE
Implement CIS AWS rule 1.17

### PR DESCRIPTION
**Pull request by orestisfl**
### Summary of your changes

Implement CIS AWS rule 1.17

### Related Issues

- Fixes: https://github.com/elastic/csp-security-policies/issues/220
- https://github.com/elastic/cloudbeat/pull/877

### Checklist

- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added the necessary README/documentation (if appropriate)

> **Note** Make sure to bump cloudbeat [`version/policy.go`](https://github.com/elastic/cloudbeat/blob/main/version/policy.go#L20) after merging this PR and drafting a new release.

### CODE REVIEW

File: bundle/compliance/cis_aws/rules/cis_1_17/test.rego [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+generate_input(roles) = {
+	"subType": "aws-policy",
+	"resource": {
+		"Arn": "arn:aws:iam::aws:policy/AWSSupportAccess",
+		"roles": roles,
+	},
+}
+
+test_violation {
```
Comment (olegsu) [inline bundle/compliance/cis_aws/rules/cis_1_17/test.rego:14]: What should be the output of
```rego
eval_fail with input as generate_input([{"RoleId": "some-id"}, {}])
```

File: bundle/compliance/cis_aws/rules/cis_1_17/rule.rego [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package compliance.cis_aws.rules.cis_1_17
+
+import data.compliance.lib.common
+import data.compliance.policy.aws_iam.data_adapter
+import future.keywords.if
+import future.keywords.in
+
+# Ensure IAM policies that allow full "*:*" administrative privileges are not attached
```
Comment (uri-weisman) [inline bundle/compliance/cis_aws/rules/cis_1_17/rule.rego:8]:
```suggestion
# Ensure a support role has been created to manage incidents with AWS Support
```

File: bundle/compliance/cis_aws/rules/cis_1_17/test.rego [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	])
+}
+
+test_pass {
+	eval_pass with input as generate_input([{"RoleId": "some-id"}])
+}
+
+test_not_evaluated {
+	not_eval with input as {}
+	not_eval with input as {"roles": []} # No subType
```
[the remark this answers is not in the document]
Comment (orestisfl) [reply bundle/compliance/cis_aws/rules/cis_1_17/test.rego:38]: It does, but redundancy in tests doesn't harm IMO

File: bundle/compliance/cis_aws/rules/cis_1_17/test.rego [hunk 4]
```diff
@@ -0,0 +1,51 @@
+package compliance.cis_aws.rules.cis_1_17
+
+import data.compliance.cis_aws.data_adapter
+import data.lib.test
+
+generate_input(roles) = {
+	"subType": "aws-policy",
```
Comment (uri-weisman) [inline bundle/compliance/cis_aws/rules/cis_1_17/test.rego:7]: I wonder if the focus should be on the role, in our findings the user would like to see which role is holding the policy.
In that case the `sub-type` should be `aws-role` and the id should be the role ARN.
wdyt?

File: bundle/compliance/cis_aws/rules/cis_1_17/test.rego (hunk 3 above)
[the remark this answers is not in the document]
Comment (olegsu) [reply bundle/compliance/cis_aws/rules/cis_1_17/test.rego:39]: Please wrap the `roles` with the `resource` object to align the structure

File: bundle/compliance/cis_aws/rules/cis_1_17/rule.rego (hunk 2 above)
Comment (orestisfl) [reply bundle/compliance/cis_aws/rules/cis_1_17/rule.rego]: Thanks :bow:

File: bundle/compliance/cis_aws/rules/cis_1_17/test.rego (hunk 4 above)
Comment (orestisfl) [reply bundle/compliance/cis_aws/rules/cis_1_17/test.rego:7]: I don't think that's possible, the rule fails when there are no roles that attach this policy. If there is no rule, we can't use it as the resource.

**Review by olegsu — APPROVED** [review]

**Pull request merged**